Add introduction (#117)
rust-unofficial/patterns: README.md

> An open source repository of design patterns and idioms in the Rust programming
> language.
> 
> ## Design Patterns Book
> 
> There is also a mdBook that you can find [here](https://rust-unofficial.github.io/patterns/).
> 

Reviewer: ```suggestion
An open source book about design patterns and idioms in the Rust programming
language that you can read [here](https://rust-unofficial.github.io/patterns/).

```

* [`#[deny(warnings)]`](anti_patterns/deny-warnings.md)

### Functional programming
* TODO currying
* TODO higher-order functions
* TODO collections

### Refactoring
* TODO refactoring from `unwrap` to `Result`
* TODO refactoring from `enum` to `Option`

> 

Reviewer: Instead of putting this in the readme, why we don't place them in proper issues so we can have organized discussions about them? In this way we have a cleaner readme.

Author: I would propose to do both actually. An link in the README.md to the issues. In this way we get attention to the issue itself on the "frontpage" of the repository, which I like and might convince people to commit more patterns. I personally don't have a big problem with the README.md being not that clean, now that we have the book rendering going on - which is the cleanest we probably can get I feel. What do you think about that?

Author: The linking of the issues in the README.md on TODOs I would prefer to do when we have the PRs merged, so we have a better overview and can use a single PR to link all issues.

Reviewer: Sure, missing patterns could be linked in the README!
Anyway once you merge this in the README, you have to explain the details in the issue. I don't know what the contributor meant with "refactoring from enum to Option" for example.
Maybe we should first create the detailed issue and once we agree that it should be done and explain what we want, we can link it to the README. What do you think?

Author: sounds good

Reviewer: So should we remove these items from the readme and create two isssues, one for functional programming and one for refactoring?

Author: done

Author: The relevant comment is here:
https://github.com/rust-unofficial/patterns/pull/43#issuecomment-258056968
